UnityProject/Assets/Scripts/Gateway/StationGateway.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Mirror;
+using UnityEngine;
+
+/// <summary>
+/// For the station Gate, only connects to a gate this script will pick
+/// </summary>
+public class StationGateway : NetworkBehaviour
+{
+	[SerializeField]
+	private SpriteRenderer[] Sprites;
+	//SpriteBaseBottom, SpriteBaseTop, SpriteBaseRightMiddle, SpriteBaseLeftMiddle, SpriteBaseRightBottom, SpriteBaseLeftBottom, SpriteBaseRightTop, SpriteBaseLeftTop, SpriteBaseCentre
+	//TODO animate centre
+
+	[SerializeField]
+	private Sprite[] Online;
+	[SerializeField]
+	private Sprite[] Offline;
+	[SerializeField]
+	private Sprite[] PowerOff;//TODO connect gateway to APC
+
+	[SerializeField]
+	private List<GameObject> Worlds = new List<GameObject>();//List of worlds available to be chosen
+
+	private GameObject SelectedWorld;// The world from the list that was chosen
+
+	private bool HasPower = true;// Not used atm
+
+	private bool IsConnected;
+
+	[SerializeField]
+	private int RandomCountBegining = 300; //Defaults to between 5 and 20 mins gate will open.
+	[SerializeField]
+	private int RandomCountEnd = 1200;
+
+	private RegisterTile registerTile;
+
+	private Matrix Matrix => registerTile.Matrix;
+
+	public string WorldName = "The Station";
+
+	private Vector3Int Position;
+
+	private string Message;
+
+	private float timeElapsedServer = 0;
+	private float timeElapsedClient = 0;
+	public float DetectionTime = 1;
+
+	[SyncVar(hook = nameof(SyncState))]
+	private bool isOn = false;
+
+	private void SyncState(bool oldVar, bool newVar)
+	{
+		isOn = newVar;
+		//do your thing
+		//all clients will be updated with this
+	}
+
+	[Server]
+	public void ServerChangeState(bool newVar)
+	{
+		isOn = newVar;
+	}
+
+	protected virtual void UpdateMe()
+	{
+		if (isServer)
+		{
+			timeElapsedServer += Time.deltaTime;
+			if (timeElapsedServer > DetectionTime && isOn == true)
+			{
+				DetectPlayer();
+				timeElapsedServer = 0;
+			}
+		}
+		else if (isClient)
+		{
```

Comment:
It's already is gonna be on the client after if(isServer) is false, no need to do if(isClient)

Reply:
Done